airflow/auth/managers/models/resource_details.py

```diff
@@ -81,6 +81,7 @@ class DagAccessEntity(Enum):
 
     AUDIT_LOG = "AUDIT_LOG"
     CODE = "CODE"
+    DAG = "DAG"
```

Comment:
I just verified, we don't need to add this. In requires_access_dag, we don't need to pass `access_entity` and it still works